advanced-integration/public/app.js

```diff
@@ -115,18 +115,13 @@ if (paypal.HostedFields.isEligible()) {
           })
             .then((res) => res.json())
             .then((orderData) => {
-              // Three cases to handle:
-              //   (1) Recoverable INSTRUMENT_DECLINED -> call actions.restart()
-              //   (2) Other non-recoverable errors -> Show a failure message
-              //   (3) Successful transaction -> Show confirmation or thank you
+              // Two cases to handle:
+              //   (1) Other non-recoverable errors -> Show a failure message
+              //   (2) Successful transaction -> Show confirmation or thank you
               // This example reads a v2/checkout/orders capture response, propagated from the server
               // You could use a different API or structure for your 'orderData'
-              var errorDetail =
+              const errorDetail =
                 Array.isArray(orderData.details) && orderData.details[0];
-              if (errorDetail && errorDetail.issue === "INSTRUMENT_DECLINED") {
-                return actions.restart(); // Recoverable state, per:
```

Comment:
Great catch. This `actions.restart()` feature works for the PayPal button to restart the flow so the buyer can try again with a different funding instrument. It does not exist in hosted fields.

Reply:
advanced-integration/views/checkout.ejs